victimsdb_lib/database.py

```diff
@@ -0,0 +1,128 @@
+
+    def javascript_vulnerabilities(self):
+        """Get VictimsDB instance containing only JavaScript vulnerabilities."""
+        ecosystem = 'javascript'
+        return VictimsDB(records={ecosystem: self._records[ecosystem]}, _ecosystem=ecosystem)
+
+    def python_vulnerabilities(self):
+        """Get VictimsDB instance containing only Python vulnerabilities."""
+        ecosystem = 'python'
+        return VictimsDB(records={ecosystem: self._records[ecosystem]}, _ecosystem=ecosystem)
+
+    def cves_for(self, name, ecosystem=None):
+        """Get list of `victimsdb_lib.Record` instances which affect given package."""
+        results = []
+        if not ecosystem:
+            ecosystem = self._ecosystem
+        for curr_ecosystem, records in self._records.items():
+            if ecosystem and curr_ecosystem != ecosystem:
+                continue
+            for record in records:
+                if record.affects(name=name):
+                    results.append(record)
+        return results
+
+    def merge(self, other_db, keep_ours=True):
+        """Merge records from `other_db` into this instance."""
+        other_records = {
+            'java': other_db.java_vulnerabilities(),
+            'javascript': other_db.javascript_vulnerabilities(),
+            'python': other_db.python_vulnerabilities()
+        }
+        for ecosystem, records in other_records.items():
+            self._merge_ecosystem(ecosystem, records, keep_ours)
+
+    def _merge_ecosystem(self, ecosystem, records, keep_ours=True):
+        for record in records:
+            if record not in self._records[ecosystem] or not keep_ours:
+                self._records[ecosystem].discard(record)
+                self._records[ecosystem].add(record)
+
+    @classmethod
+    def from_dir(cls, db_dir):
+        """Build database from directory."""
+        records = {
+            'java': set(),
+            'javascript': set(),
+            'python': set()
+        }
+
+        for ecosystem in records.keys():
+            db_dir_lang = os.path.join(db_dir, ecosystem)
+            for dirpath, dirnames, filenames in os.walk(db_dir_lang):
+                for filename in filenames:
+                    if not filename.endswith(('.yaml', '.yml')):
+                        continue
+                    try:
+                        fullpath = os.path.join(dirpath, filename)
+                        record = Record.from_file(fullpath, ecosystem)
+                        records[ecosystem].add(record)
+                    except ParseError as e:
+                        _logger.warning(str(e))
+        return cls(records=records)
+
+    @classmethod
+    def from_git_url(cls, git_url):
+        """Build database from GIT URL."""
+        with tempfile.TemporaryDirectory() as temp_dir:
+            subprocess.check_call(['git', 'clone', '--single-branch', git_url, temp_dir])
```

Comment:
`--depth=1` is another good option together with `--single-branch`

Reply:
Ah, right! Thanks. I knew there was such option, but from some reason I thought it was `--shallow` or something like that. Which didn't work.